napari/utils/colormaps/colormap_utils.py

```diff
@@ -120,6 +120,64 @@
     for name, (display_name, value) in inverse_cmaps.items()
 }
 
+# This is the hardcoded default colormap that is produced by
+# napari code up to version 0.4.18.
+# To keep user experience consistent,
+# we keep this colormap as the default for now.
+CLASSICAL_LABEL_COLORMAP = np.array(
+    [
+        [0.56607443, 0.41737616, 0.35140616, 1.0],
+        [0.47063142, 0.14611654, 0.027308635, 1.0],
+        [0.35923997, 0.83787304, 0.9764158, 1.0],
+        [0.57314, 0.53869504, 0.9130728, 1.0],
+        [0.42733493, 0.009019371, 0.75742406, 1.0],
+        [0.28053862, 0.22821146, 0.6264092, 1.0],
+        [0.67241573, 0.92709625, 0.5439105, 1.0],
+        [0.5636559, 0.68220073, 0.7002792, 1.0],
+        [0.5277779, 0.5672113, 0.6043446, 1.0],
+        [0.9987752, 0.9686924, 0.10985588, 1.0],
+        [0.97181, 0.27003965, 0.23497851, 1.0],
+        [0.013270683, 0.30796805, 0.22586866, 1.0],
+        [0.4466232, 0.7004097, 0.38028476, 1.0],
+        [0.41520673, 0.5876629, 0.29260996, 1.0],
+        [0.23717083, 0.10942184, 0.5298238, 1.0],
+        [0.18114366, 0.28164777, 0.06577954, 1.0],
+        [0.36103004, 0.7199946, 0.8734126, 1.0],
+        [0.28011563, 0.618632, 0.46517712, 1.0],
+        [0.27922845, 0.26164937, 0.34803712, 1.0],
+        [0.9410924, 0.3381435, 0.39963824, 1.0],
+        [0.7241666, 0.3146531, 0.48499578, 1.0],
+        [0.6544138, 0.16842732, 0.3960731, 1.0],
+        [0.94641864, 0.70651007, 0.9280277, 1.0],
+        [0.297329, 0.12059682, 0.12751216, 1.0],
+        [0.7076346, 0.77454394, 0.24515858, 1.0],
+        [0.6125465, 0.37994882, 0.18645138, 1.0],
+        [0.5435218, 0.26637766, 0.10691992, 1.0],
+        [0.84836274, 0.7452487, 0.6049582, 1.0],
+        [0.70704526, 0.4909478, 0.74280363, 1.0],
+        [0.6511858, 0.3691862, 0.64505434, 1.0],
+        [0.3783068, 0.47851342, 0.20781435, 1.0],
+        [0.7586575, 0.8936002, 0.34182736, 1.0],
+        [0.8462168, 0.586672, 0.3367908, 1.0],
+        [0.6208729, 0.53013545, 0.43864474, 1.0],
+        [0.0013958708, 0.17632863, 0.92381346, 1.0],
+        [0.7957948, 0.905851, 0.8813487, 1.0],
+        [0.8830641, 0.5477784, 0.12084796, 1.0],
+        [0.47108608, 0.9424037, 0.006252195, 1.0],
+        [0.48402575, 0.5575866, 0.05954063, 1.0],
+        [0.9018971, 0.86706156, 0.7006243, 1.0],
+        [0.26258108, 0.8489664, 0.6514427, 1.0],
+        [0.27836594, 0.73216563, 0.55689317, 1.0],
+        [0.6879435, 0.29696935, 0.94934344, 1.0],
+        [0.6922087, 0.050221473, 0.24815048, 1.0],
+        [0.96522236, 0.53059846, 0.65577245, 1.0],
+        [0.79853475, 0.19963147, 0.7806417, 1.0],
+        [0.50714034, 0.30618754, 0.26792714, 1.0],
+        [0.9122511, 0.7068656, 0.42555025, 1.0],
+        [0.23882896, 0.1612251, 0.2637005, 1.0],
+    ]
+)[np.argsort(np.random.default_rng(500).permutation(49))]
```

Comment:
Wait why is this shuffled here and why is the seed 500? The idea would be for labels 1-49 to map to the same colours as they do now. For that, I think what you want to do is something like:

```python
colormap = label_colormap(49)  # or whatever it is now
classical_colors = colormap.map(np.arange(100))
```

I think? And I would keep it as code rather than hardcoded numbers... What do you think?

Reply:
But @brisvag is right: the goal here is to preserve the look of the first few labels (say up to 100, not sure what a good number is here, but at a minimum 1-10) of the napari colormap. If people do:

```python
viewer = napari.Viewer()
viewer.add_labels(np.arange(49).reshape((7, 7)))
```

they should get the same visual result both before and after this PR.

Reply:
the suffle is to have reversed permutation for this part of code: https://github.com/napari/napari/pull/6179/files#diff-93b06ecc55258074223c21034a86e7fbaaffe5157565c04c336baec8260a7777R502-R505

And 500 is because the default seed is 0.5 and I multiply it by 1000

Reply:
We need to make sure not only that this is identical before and after this PR:

```py
viewer = napari.Viewer()
viewer.add_labels(np.arange(49).reshape((7, 7)))
```

but more generally that the *first* 49 labels remain the same, while the rest might change (i.e: also when we have > 49 labels, the first 49 are generated with this list):

```py
viewer = napari.Viewer()
viewer.add_labels(np.arange(56).reshape((7, 8)))
```